### PR TITLE
OAK-9891: Removal (purge) of version of a node does not remove associated labels

### PR DESCRIPTION
**Pull request by ArunOnCloud**
pull request for issue : OAK-9891
mail thread link: https://lists.apache.org/thread/b1g2f3cxxfpfnqxzr3oz20lyp8676tnm

### CODE REVIEW

**Comment by ArunOnCloud** [comment]
@mreutegg can you please review it.

File: oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java [hunk 1]
```diff
@@ -159,4 +162,48 @@ public void testRemoveVHR() throws RepositoryException {
...
+        Node n = testRootNode.addNode(nodeName1, testNodeType);
+        n.addMixin(mixVersionable);
+        superuser.save();
+
+        VersionManager vm = superuser.getWorkspace().getVersionManager();
+        for (int i = 0; i < createVersions; i++) {
+            vm.checkout(n.getPath());
+            vm.checkin(n.getPath());
+        }
+        superuser.save();
```
Comment (mreutegg) [inline oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java]: This is unnecessary. Checkin and checkout are workspace operations and do not need to be saved.

File: oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java [hunk 2]
```diff
@@ -159,4 +162,48 @@ public void testRemoveVHR() throws RepositoryException {
...
         assertFalse("VersionHistory node should have disappeared", superuser.itemExists(vhrpath));
     }
+
+    public void testRemoveVersionLabelWithRemovalOfVersion() throws RepositoryException{
+        int createVersions = 3;
+        Node n = testRootNode.addNode(nodeName1, testNodeType);
+        n.addMixin(mixVersionable);
+        superuser.save();
+
+        VersionManager vm = superuser.getWorkspace().getVersionManager();
```
Comment (mreutegg) [inline oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java]: There is already a `VersionManager` available as an instance variable `versionManager`.

File: oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java [hunk 3]
```diff
@@ -159,4 +162,48 @@ public void testRemoveVHR() throws RepositoryException {
...
+        }
+        int versonLabelCount = 3;
+        List<String> versionLabels = new ArrayList<>();
+        for(int i = 0; i < versonLabelCount; i++) {
+            String labelName = "Label_" + versionName + "_" + i;
+            vhr.addVersionLabel(versionName, labelName,false);
+            versionLabels.add(labelName);
+        }
+
+        superuser.save();
```
Comment (mreutegg) [inline oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java]: Again, this save() call is unnecessary. `VersionHistory.addVersionLabel()` is a workspace operation.

File: oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java [hunk 4]
```diff
@@ -159,4 +162,48 @@ public void testRemoveVHR() throws RepositoryException {
...
+        vhr.removeVersion(versionName);
+        superuser.save();
```
Comment (mreutegg) [inline oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java]: Similar as above, `removeVersion()` is again a workspace operation.

File: oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java [hunk 5]
```diff
@@ -159,4 +162,48 @@ public void testRemoveVHR() throws RepositoryException {
...
+
+        for(String label : versionLabels) {
+            assertEquals("version label should not exist", false, vhr.hasVersionLabel(label));
```
Comment (mreutegg) [inline oak-jcr/src/test/java/org/apache/jackrabbit/oak/jcr/version/VersionHistoryTest.java]: This assertion would be easier to read if it was using `assertFalse()`.

**Review by mreutegg — CHANGES_REQUESTED** [review]

**Review by mreutegg — APPROVED** [review]
Looks good to me.

**Pull request merged**